services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/DpmServiceImpl.java

```diff
@@ -79,14 +94,15 @@ private DpmServiceImpl(Config globalConfig) {
 
     private UnicastGoalState buildUnicastGoalState(NetworkConfiguration networkConfig, String hostIp,
                                                    List<InternalPortEntity> portEntities,
-                                                   MulticastGoalState multicastGoalState) throws Exception {
+                                                   MulticastGoalState multicastGoalState,
+                                                   ZetaPortGoalState zetaPortGoalState) throws Exception {
```

Comment:
The current way of injecting zeta-specific variables are a bit intrusive to OVS specific codes.

I would recommend to try to use a different set of methods from OVS (in the same DPMServiceImpl), or create a new DPMServiceImpleForZeta, which is another option.

Reply:
Zeta feature and dpm service are decoupled, there are only two methods in `DpmServiceImpl.java` have some coherent codes: `processPortConfiguration` and `doCreatePortConfiguration`.